### PR TITLE
otel events on evaluation

### PR DESCRIPTION
**Pull request by passichenko**
Simplified version of https://github.com/lekkodev/go-sdk/pull/63

This is opt-in feature, it has to be enabled via `WithOtelTracing()` option.
If enabled, for evaluation try to add an event to the active span. If OTel is not set up properly or there is no active span, this will be a no-op. In the event we log primitive values (bool, int, float, string), json and proto are not logged to avoid cost of serialization.

### CODE REVIEW

**Comment by passichenko** [comment]
* **#119** <a href="https://app.graphite.dev/github/pr/lekkodev/go-sdk/119?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @passichenko and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: client/cached_provider.go [hunk 1]
```diff
@@ -150,17 +170,27 @@ func (cp *cachedProvider) GetJSON(ctx context.Context, key string, namespace str
...
 }
 
 func (cp *cachedProvider) GetString(ctx context.Context, key string, namespace string) (string, error) {
 	dest := &wrapperspb.StringValue{}
-	if err := cp.store.Evaluate(key, namespace, fromContext(ctx), dest); err != nil {
+	cfg, err := cp.store.Evaluate(key, namespace, fromContext(ctx), dest)
+	if err != nil {
 		return "", err
 	}
+	cp.otel.addTracingEvent(ctx, key, dest.GetValue(), cfg.CommitSHA)
```
Comment (passichenko) [inline client/cached_provider.go:194]: one can argue that logging strings could be dangerous, but let's worry about it later (we can have a flag for this)

File: client/provider.go [hunk 2]
```diff
@@ -205,3 +208,44 @@ func (o *repositoryKeyOption) apply(cfg *providerConfig) {
...
+func (o *otelTracing) addTracingEvent(ctx context.Context, key, stringValue, version string) {
+	if o == nil {
+		return
+	}
+
+	span := trace.SpanFromContext(ctx)
+
+	var attributes []attribute.KeyValue
+	if len(stringValue) > 0 {
+		attributes = append(attributes, attribute.String(fmt.Sprintf("config.%s.string_value", key), stringValue))
```
Comment (DavidSGK) [inline client/provider.go:237]: Bit unclear on the attribute naming choices, is it intentional to have both `config.xxx....` and `lekko.xxx`?

**Review by DavidSGK — APPROVED** [review]

File: client/provider.go (hunk 2 above)
Comment (passichenko) [reply client/provider.go:237]: Good catch, I forgot to update that. Agree, it should be consistent.

**Pull request merged**